src/fileparse.py

```diff
@@ -0,0 +1,66 @@
+import csv
+import logging
+from pathlib import Path
+from typing import Any, List, TextIO, Tuple, Union, Callable, Dict, cast
+
+
+def parse_csv(
+    file_path_or_buffer: Union[str, Path, TextIO],
+    select: List[str] = None,
+    types: List[Callable] = None,
+    has_headers: bool = True,
+    delimiter: str = ",",
+    silence_errors=False,
```

Comment:
You forgot the static typing:
```suggestion
    silence_errors: bool = False,
```

Also we general, this kind of parameter is named `verbose` cf sklearn's [IsolationForest](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.IsolationForest.html) where verbose can be:
- an integer if you have different level of verbosity: 0 == nothing logged, 1 == some stuff are logged, 2 == more stuff are logged (it really depends on the use case)
- an boolean: False == log nothing, True == log

Reply:
Changed to `verbose`.